cstar/tests/unit_tests/roms/test_roms_simulation.py

```diff
@@ -359,93 +365,49 @@ def test_default_codebase_assignment(self, tmp_path):
         )
         assert sim.marbl_codebase.checkout_target == "marbl0.45.0"
 
-    def test_invalid_surface_forcing_type(self, tmp_path):
-        """Ensure a `TypeError` is raised when `surface_forcing` is not a list of
-        `ROMSSurfaceForcing` instances.
-
-        This test verifies that the `ROMSSimulation` constructor enforces type safety
-        by checking that `surface_forcing` is a list containing only `ROMSSurfaceForcing`
-        objects. Passing an invalid list should result in a `TypeError`.
-
-        Assertions
-        ----------
-        - A `TypeError` is raised when `surface_forcing` contains elements that are
-          not `ROMSSurfaceForcing` instances, including an expected message substring.
-
-        Mocks & Fixtures
-        ----------------
-        - `tmp_path` : A temporary directory fixture provided by `pytest` for
-          safely testing file system interactions.
-        """
-
-        with pytest.raises(
-            TypeError, match="must be a list of ROMSSurfaceForcing instances"
-        ):
-            ROMSSimulation(
-                name="test",
-                directory=tmp_path,
-                discretization=ROMSDiscretization(time_step=60),
-                codebase=ROMSExternalCodeBase(),
-                runtime_code=AdditionalCode(location="some/dir"),
-                compile_time_code=AdditionalCode(location="some/dir"),
-                start_date="2012-01-01",
-                end_date="2012-01-02",
-                valid_start_date="2012-01-01",
-                valid_end_date="2012-01-02",
-                surface_forcing=[
-                    "list",
-                    "of",
-                    "strings",
-                    "instead",
-                    "of",
-                    "ROMSSurfaceForcing",
-                    "instances",
-                ],
-            )
-
-    def test_invalid_boundary_forcing_type(self, tmp_path):
-        """Ensure a `TypeError` is raised when `boundary_forcing` is not a list of
-        `ROMSBoundaryForcing` instances.
-
-        This test verifies that the `ROMSSimulation` constructor enforces type safety
-        by ensuring that `boundary_forcing` is a list containing only `ROMSBoundaryForcing`
-        objects. Passing an invalid list should result in a `TypeError`.
+    @pytest.mark.parametrize(
+        "argname, expected_type_name",
+        [
+            ("surface_forcing", "ROMSSurfaceForcing"),
+            ("boundary_forcing", "ROMSBoundaryForcing"),
+            ("forcing_corrections", "ROMSForcingCorrections"),
+        ],
+    )
+    def test_lists_of_input_datasets_raise_if_invalid(
+        self, tmp_path, argname, expected_type_name
```

Comment:
i'm a fan of typehints in tests. i know it's not consistent now, but consider adding them to make the tests easier to check with static analysis, later